### ISSUE TITLE
Add option to disable static profile page

### ISSUE THREAD

**Issue by ashkitten** [glitch-soc/mastodon#445]
Should just disable the HTML page for the profile, not specific toots

* * * *

- [x] I searched or browsed the repo’s other issues to ensure this is not a duplicate.
- [ ] This bug happens on a [tagged release](https://github.com/tootsuite/mastodon/releases) and not on `master` (If you're a user, don't worry about this).

**Comment by ThibG** [comment]
Would that also apply to the `ActivityPub` endpoint? i.e., it would list no toot?

**Comment by yipdw** [comment]
My understanding is that this only applies to the HTML output of `AccountsController#show`, and not the Atom or JSON outputs of that action, with the idea being to discourage discoverability via web browsers.

I think we could also disable the Atom and JSON outputs without affecting the ability to follow / request follow and send/receive updates, though?  It'd still be possible to webfinger an account in that state.

**Comment by nightpool** [comment]
no, webfinger is just a reference to the json output. it would completely
break federation (that's what countersoc did, on accident)

**Comment by yipdw** [comment]
Ah, okay.  Yeah, then I guess it'd just be the HTML page.

**Comment by yipdw** [comment]
I'll do just the HTML for now -- I understand the effects of doing that much better than disabling the JSON outputs (obviously).  If it turns out we need more, we can do more.
